audits/supersimple.py

```diff
@@ -0,0 +1,175 @@
+
+    def get_sample_sizes(self, contests, margins, total_ballots, reported_results, sample_results):
+        """
+        Computes initial sample sizes parameterized by likelihood that the
+        initial sample will confirm the election result, assuming no
+        discrepancies.
+
+        Inputs:
+            total_ballots  - the total number of ballots cast in the election
+            sample_results - if a sample has already been drawn, this will
+                             contain its results. 
+
+        Outputs:
+            samples - dictionary mapping confirmation likelihood to sample size:
+                    {
+                       contest1:  { 
+                            likelihood1: sample_size,
+                            likelihood2: sample_size,
+                            ...
+                        },
+                        ...
+                    }
+        """
+
+        rho = -math.log(self.risk_limit)/((1/(2*self.gamma)) + self.l*math.log(1 - 1/(2*self.gamma))) 
+
+        diluted_margin = self.compute_diluted_margin(contests, margins, total_ballots)
+
+        return math.ceil(rho/diluted_margin) 
+
+    def compute_risk(self, contests, margins, cvrs, sample_cvr):
+        """
+        Computes the risk-value of <sample_results> based on results in <contest>.
+
+        Inputs: 
+            contests       - the contests and results being audited
+            margins        - the margins for the contest being audited
+            cvrs           - mapping of ballot_id to votes:
+                    {
+                        'ballot_id': {
+                            'contest': {
+                                'candidate1': 1,
+                                'candidate2': 0,
+                                ...
+                            }
+                        ...
+                    }
+
+            sample_cvr - the CVR of the audited ballot  
+                    {
+                        'ballot_id': {
+                            'contest': {
+                                'candidate1': 1,
+                                'candidate2': 0,
+                                ...
+                            }
+                        ...
+                    }
+
+        Outputs:
+            measurements    - the p-value of the hypotheses that the election
+                              result is correct based on the sample, for each winner-loser pair. 
+            confirmed       - a boolean indicating whether the audit can stop
+        """
+        
+        p = 1
+
+
+        diluted_margin = self.compute_diluted_margin(contests, margins, len(cvrs))
+        V = diluted_margin*len(cvrs)
+
+        U = 2*self.gamma/diluted_margin
+
+        lowest_p = 1
+        result = False
+        for ballot in sample_cvr:
+            e_r = 0
+
+            for contest in contests:
+                if contest not in sample_cvr[ballot]:
+                    # TODO I think this is right
```

Comment:
At least conceptually, yes -- if the contest isn't on the ballot, there's no effect on the p value.